components/tabs/src/storage.rs

```diff
@@ -293,18 +295,40 @@ impl TabsStorage {
                 return crts;
             }
         };
+        // building this in to a hashset seems a little over the top?
+        let pending_close_inactive_result: Result<Vec<String>> = conn.query_rows_and_then_cached(
+            "SELECT device_id FROM remote_tab_commands WHERE command = :command_close_inactive",
+            rusqlite::named_params! {":command_close_inactive": CommandKind::CloseInactiveTabs as u8},
+            |row| {
+                Ok(row.get::<_, String>(0)?) // device_id
+            },
+        );
+        let pending_close_inactive = match pending_close_inactive_result {
+            Ok(rows) => rows,
+            Err(e) => {
+                error_support::report_error!(
+                    "tabs-read-remote",
+                    "Failed to read database for remote inactive closes: {}",
+                    e
+                );
+                Default::default()
+            }
+        };
+
         // Check if any of the client records that were passed in have urls that the user closed
         // This means that they requested to close those tabs but those devices have not yet got
         // actually closed the tabs
```

Comment:
I do wonder if there's a way we could generalize this kind of "commands subsuming other commands" in the future—but no need to do that in this PR!